api/src/main/java/io/kafbat/ui/client/RetryingKafkaConnectClient.java

```diff
@@ -56,10 +57,23 @@ private static <T> Flux<T> withRetryOnConflict(Flux<T> publisher) {
     return publisher.retryWhen(conflictCodeRetry());
   }
 
+  // Adapted from https://github.com/apache/kafka/blob/a0a501952b6d61f6f273bdb8f842346b51e9dfce/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/entities/ErrorMessage.java
+  // We only need this class from the connect runtime and adding the entire dependency seems excessive
+  // We also only need the `message` property
+  private record ErrorMessage(@JsonProperty("message") String message) {
+  }
+
   private static <T> Mono<T> withBadRequestErrorHandling(Mono<T> publisher) {
     return publisher
-        .onErrorResume(WebClientResponseException.BadRequest.class, e ->
-            Mono.error(new ValidationException("Invalid configuration")))
+        .onErrorResume(WebClientResponseException.BadRequest.class, e -> {
+          final var errorMessage = e.getResponseBodyAs(ErrorMessage.class);
+
+          if (errorMessage != null && errorMessage.message() != null) {
+            return Mono.error(new ValidationException(errorMessage.message()));
```

Comment:
I think that the raw error from Kafka Connect is enough but one other idea is to append "Invalid Configuration" if needed